tests/distributed/test_distributed_sampling.py

```diff
@@ -132,10 +132,69 @@ def test_rpc_sampling_shuffle():
         check_rpc_sampling_shuffle(Path(tmpdirname), 2)
         check_rpc_sampling_shuffle(Path(tmpdirname), 1)
 
+def start_in_subgraph_client(rank, tmpdir, disable_shared_mem, nodes):
+    import dgl
+    gpb = None
+    if disable_shared_mem:
+        _, _, _, gpb = load_partition(tmpdir / 'test_in_subgraph.json', rank)
+    dist_graph = DistGraph("rpc_sampling_ip_config.txt", "test_in_subgraph", gpb=gpb)
+    sampled_graph = dgl.distributed.in_subgraph(dist_graph, nodes)
+    dgl.distributed.shutdown_servers()
+    dgl.distributed.finalize_client()
+    return sampled_graph
+
+
+def check_rpc_in_subgraph(tmpdir, num_server):
+    ip_config = open("rpc_sampling_ip_config.txt", "w")
+    for _ in range(num_server):
+        ip_config.write('{} 1\n'.format(get_local_usable_addr()))
+    ip_config.close()
+
+    g = CitationGraphDataset("cora")[0]
+    g.readonly()
+    num_parts = num_server
+
+    partition_graph(g, 'test_in_subgraph', num_parts, tmpdir,
+                    num_hops=1, part_method='metis', reshuffle=False)
+
+    pserver_list = []
+    ctx = mp.get_context('spawn')
+    for i in range(num_server):
+        p = ctx.Process(target=start_server, args=(i, tmpdir, num_server > 1, 'test_in_subgraph'))
+        p.start()
+        time.sleep(1)
+        pserver_list.append(p)
+
+    nodes = [0, 10, 99, 66, 1024, 2008]
+    time.sleep(3)
+    sampled_graph = start_in_subgraph_client(0, tmpdir, num_server > 1, nodes)
+    for p in pserver_list:
+        p.join()
+
+    src, dst = sampled_graph.edges()
+    g = dgl.as_heterograph(g)
+    assert sampled_graph.number_of_nodes() == g.number_of_nodes()
+    subg1 = dgl.in_subgraph(g, nodes)
+    src1, dst1 = subg1.edges()
+    assert np.all(np.sort(F.asnumpy(src)) == np.sort(F.asnumpy(src1)))
+    assert np.all(np.sort(F.asnumpy(dst)) == np.sort(F.asnumpy(dst1)))
+    eids = g.edge_ids(src, dst)
+    assert np.array_equal(
+        F.asnumpy(sampled_graph.edata[dgl.EID]), F.asnumpy(eids))
```

Comment:
Better to compare the result with dgl.in_subgraph

Reply:
the order of the edges in the subgraphs isn't the same. we have to sort them, in which case, we cannot check if the edge Ids are associated with the right endpoint nodes. this check can check the correctness of the results as well.